File: 2946-matrix-similarity-after-cyclic-shifts/2946-matrix-similarity-after-cyclic-shifts.cpp
class Solution {
public:
    bool areSimilar(vector<vector<int>>& mat, int k) {
        
        k = k % mat[0].size() ;

        for( int i = 0 ; i < mat.size() ; i++ ){
            
            vector<int> row = mat[i] ;

            if( i % 2 == 1 ){
                for( int j = 0 ; j < mat[0].size() ; j++ ){
                    row[ ( j + k ) % mat[0].size() ] = mat[i][j] ;
                }
            }
            else{
                
                int index = 0 ;
                for( int j = k ; j < mat[0].size() ; j++ ){
                    row[index] = mat[i][j] ;
                    index ++ ;
                }
                for( int j = 0 ; j < k ; j++ ){
                    row[index] = mat[i][j] ;
                    index ++ ;
                }

            }

            if( row != mat[i] ) return false ;
        }

        return true ;
    }
};

File: 2946-matrix-similarity-after-cyclic-shifts/2946-matrix-similarity-after-cyclic-shifts.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {boolean}
 */
var areSimilar = function(mat, k) {
    
    k = k % mat[0].length ;

    for( let i = 0 ; i < mat.length ; i++ ){

        let arr = [ ...mat[i] ]

        if( i % 2 == 1 ){
            for( let j = 0 ; j < mat[0].length ; j++ ){
                arr[ ( j + k ) % mat[0].length ] = mat[i][j] ;
            }
        }
        else{
            let index = 0 ;
            for( let j = k ; j < mat[0].length ; j++ ){
                arr[index] = mat[i][j] ;
                index ++ ;
            }
            for( let j = 0 ; j < k ; j++ ){
                arr[index] = mat[i][j] ;
                index ++ ;
            }
        }

        if (JSON.stringify(arr) !== JSON.stringify(mat[i])) return false;
    }

    return true ;
};